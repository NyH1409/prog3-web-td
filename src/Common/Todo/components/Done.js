import { TaskAlt } from "@mui/icons-material";
import { Card, CardContent, Paper, Typography } from "@mui/material";
import { pink, red } from "@mui/material/colors";
import React from "react";

export function Done(props) {
    const {done, title} = props;

    return (
        <>
        <Paper elevation={3} className="done-container" sx={{ width: 275 }}>
        <Typography variant="h5" sx={{ textAlign: "center", fontWeight:"bold", color: red[300] }}>DONE</Typography>
            {
                done?.map((elt)=>(
                    <Card sx={{ width : 275, margin : "8px 0" }} elevation={5} key={elt?.id}>
                        <CardContent>
                            <TaskAlt sx={{ color: pink[500]}} />
                            <Typography data-testid="title" color="text.secondary">{title}</Typography>
                            <Typography sx={{ mb: 1.5 }} color="blue">{new Date().toISOString().slice(0, 10)}</Typography>
                            <Typography data-testid="element-title" variant="body2" sx={{paddingRight: 2}}>{elt?.title}<br /></Typography>
                        </CardContent>
                    </Card>
                ))
            }
        </Paper>
        </>
    )
}