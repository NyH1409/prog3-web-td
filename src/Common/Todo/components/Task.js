import { Receipt } from "@mui/icons-material";
import { Card, CardActions, CardContent, FormControlLabel, FormGroup, Paper, Typography } from "@mui/material";
import { pink, red } from "@mui/material/colors";
import React from "react";

export function Task(props) {
    const {data, doneTask, title} = props;
    return (
        <>
        <Paper elevation={3} className="todo-container" 
        sx={{ width: 275, minHeight: 400, bgcolor: red[300], }}>
            <Typography variant="h5" sx={{ textAlign: "center" , fontWeight: "bold", color: "white" }}>TODO</Typography>
            {
                data?.map((elt)=>(
                    <Card sx={{ width : 275, margin : "8px 0" }} elevation={5}>
                        <CardContent>
                            <Receipt sx={{ color: pink[500]}} />
                            <Typography color="text.secondary">{title}</Typography>
                            <Typography sx={{ mb: 1.5 }} color="blue">{new Date().toISOString().slice(0, 10)}</Typography>
                            <Typography variant="body2" sx={{paddingRight: 2}}>{elt?.title}<br /></Typography>
                        </CardContent>
                        <CardActions sx={{ paddingLeft: 2}}>
                            <FormGroup>
                                <FormControlLabel control={<input className="to-check" type="checkbox" onClick={()=>doneTask(elt)}/>} label="Check if done"/>
                            </FormGroup>
                        </CardActions>
                    </Card>
                ))
            }
        </Paper>
        </>
    )
}