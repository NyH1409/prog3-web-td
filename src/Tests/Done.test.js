import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Done } from "../Common/Todo/components/Done";
import { task } from "./TestUtils/TestUtils";

describe("Test done component", ()=>{
    it("Render correctly", ()=>{
        const component = renderer.create(
            <Done done={task} title="Done title" />
        )
        const tree = component.toJSON();
        
        expect(tree).toMatchSnapshot();
        expect(task).toMatchSnapshot([{
            id: expect.any(Number),
            title: "Test title"
        }]);
        expect("Done title").toMatchSnapshot();
    })

    it("Read component props ok", ()=>{
        const { getByTestId } = render(
            <Done done={task} title="Done title" />
        )

        expect(getByTestId("title").textContent).toBe("Done title");
        expect(getByTestId("element-title").textContent).toBe("Test title");
    })
})