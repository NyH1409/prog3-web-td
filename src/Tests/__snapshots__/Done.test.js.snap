// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test done component Render correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 done-container css-8f5jog-MuiPaper-root"
>
  <h5
    className="MuiTypography-root MuiTypography-h5 css-1bq8dcc-MuiTypography-root"
  >
    DONE
  </h5>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 MuiCard-root css-29ol60-MuiPaper-root-MuiCard-root"
  >
    <div
      className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1b32eso-MuiSvgIcon-root"
        data-testid="TaskAltIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M22 5.18 10.59 16.6l-4.24-4.24 1.41-1.41 2.83 2.83 10-10L22 5.18zm-2.21 5.04c.13.57.21 1.17.21 1.78 0 4.42-3.58 8-8 8s-8-3.58-8-8 3.58-8 8-8c1.58 0 3.04.46 4.28 1.25l1.44-1.44C16.1 2.67 14.13 2 12 2 6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10c0-1.19-.22-2.33-.6-3.39l-1.61 1.61z"
        />
      </svg>
      <p
        className="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
        data-testid="title"
      >
        Done title
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 css-16ezf3a-MuiTypography-root"
      >
        2023-02-06
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2 css-14o3s5-MuiTypography-root"
        data-testid="element-title"
      >
        Test title
        <br />
      </p>
    </div>
  </div>
</div>
`;

exports[`Test done component Render correctly 2`] = `
Array [
  Object {
    "id": Any<Number>,
    "title": "Test title",
  },
]
`;

exports[`Test done component Render correctly 3`] = `"Done title"`;
